version: '3.9'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - "8000:8000"
    volumes:
      - ./media/images/uploads:/app/media/images/uploads
    depends_on:
      - db
    environment:
      - DJANGO_DB_HOST=${DJANGO_DB_HOST}
      - DJANGO_DB_NAME=${DJANGO_DB_NAME}
      - DJANGO_DB_USER=${DJANGO_DB_USER}
      - DJANGO_DB_PASS=${DJANGO_DB_PASS}
      - DJANGO_DB_PORT=${DJANGO_DB_PORT}
      - DJANGO_DEBUG=${DJANGO_DEBUG}
    command: [ "bash", "-c", "python manage.py migrate && uwsgi --ini uwsgi.ini"]
    networks:
      - app

  db:
    # образ, используемый для контейнера
    image: postgres:14-alpine
    container_name: ${DJANGO_DB_HOST}
    # переменные окружения для БД
    environment:
      - POSTGRES_DB=${DJANGO_DB_NAME}
      - POSTGRES_USER=${DJANGO_DB_USER}
      - POSTGRES_PASSWORD=${DJANGO_DB_PASS}
    volumes:
      - ./data/postgresql:/var/lib/postgresql/data
    expose:
      - 5432
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $DJANGO_DB_USER -d $DJANGO_DB_NAME" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
        - app

networks:
  app:
    driver: bridge
